hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go

```diff
@@ -184,14 +193,62 @@ func (a *Azure) ReconcileCredentials(ctx context.Context, c client.Client, creat
 	return nil
 }
 
-func (a *Azure) ReconcileSecretEncryption(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+func (a Azure) ReconcileSecretEncryption(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+	return nil
+}
+
+func (a Azure) CAPIProviderPolicyRules() []rbacv1.PolicyRule {
+	return nil
+}
+
+func (a Azure) DeleteCredentials(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
 	return nil
 }
 
-func (a *Azure) CAPIProviderPolicyRules() []rbacv1.PolicyRule {
+func reconcileAzureCluster(azureCluster *capiazure.AzureCluster, hcluster *hyperv1.HostedCluster, apiEndpoint hyperv1.APIEndpoint, controlPlaneNamespace string) error {
+	if azureCluster.Annotations == nil {
+		azureCluster.Annotations = map[string]string{}
+	}
+
+	azureCluster.Annotations[capiv1.ManagedByAnnotation] = "external"
+
+	azureCluster.Spec.Location = hcluster.Spec.Platform.Azure.Location
+	azureCluster.Spec.ResourceGroup = hcluster.Spec.Platform.Azure.ResourceGroupName
+	azureCluster.Spec.NetworkSpec.Vnet.ID = hcluster.Spec.Platform.Azure.VnetID
+	azureCluster.Spec.NetworkSpec.Vnet.Name = hcluster.Spec.Platform.Azure.VnetName
+	azureCluster.Spec.NetworkSpec.Vnet.ResourceGroup = hcluster.Spec.Platform.Azure.ResourceGroupName
+	azureCluster.Spec.SubscriptionID = hcluster.Spec.Platform.Azure.SubscriptionID
+
+	azureCluster.Spec.ControlPlaneEndpoint = capiv1.APIEndpoint{
+		Host: apiEndpoint.Host,
+		Port: apiEndpoint.Port,
+	}
+
+	azureCluster.Status.Ready = true
+
+	azureCluster.Spec.IdentityRef = &corev1.ObjectReference{Name: hcluster.Name, Namespace: controlPlaneNamespace}
+
 	return nil
 }
 
-func (a *Azure) DeleteCredentials(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+func reconcileAzureClusterIdentity(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, azureClusterIdentity *capiazure.AzureClusterIdentity, controlPlaneNamespace string) error {
+	var userCloudCredSecret corev1.Secret
```

Comment:
you could as well declare the secret and then infer the object key
```
	secret := &corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{
			Name:                       "x",
			Namespace:                  "y",
		},
	}
	objectKey := client.ObjectKeyFromObject(secret)
```

Reply:
For my own edification, is there a reason to prefer one method over the other?

Reply:
the above is a common pattern in our code base where you only declare a variable once.
If you prefer to keep the current code please rename userCloudCredSecret to credentialsSecret, and secretName to secretObjectKey/secretNamespacedName